components/calendar/calendar.js

```diff
@@ -0,0 +1,268 @@
+import React from 'react';
+import { getClassNamesWithMods, getDataAttributes } from '../_helpers';
+import DaysPanel from './panels/days';
+import calendarConstants from './constants/calendar';
+
+const {
+  CALENDAR_MOVE_TO_NEXT,
+  CALENDAR_MOVE_TO_PREVIOUS,
+  CALENDAR_SELECTION_TYPE_RANGE,
+} = calendarConstants;
+
+const { PropTypes } = React;
+
+export default class Calendar extends React.Component {
+  constructor(props) {
+    super();
+
+    const { initialDates, maxDate, minDate } = props;
+    const maxLimit = maxDate ? new Date(maxDate) : null;
+    const renderDate = (initialDates && initialDates.length) ? new Date(initialDates[0]) : new Date();
+    const selectedDates = initialDates ? initialDates.map(item => (item ? new Date(item) : null)) : [null, null];
```

Comment:
two inlined ternary operators with arrow function - hard to read\understand, please consider simplifying and\or extract